### PR TITLE
Jvenstad/integration test deployments with new deploy path

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/stubs/MockTesterCloud.java [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
 import java.util.List;
+import java.util.Optional;
 import java.util.stream.Collectors;
 
 import static com.yahoo.vespa.hosted.controller.api.integration.deployment.TesterCloud.Status.NOT_STARTED;
 import static com.yahoo.vespa.hosted.controller.api.integration.deployment.TesterCloud.Status.RUNNING;
 
 public class MockTesterCloud implements TesterCloud {
 
+    private final NameService cNames;
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/stubs/MockTesterCloud.java]: s/cNames/nameService/

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java [hunk 2]
```diff
@@ -472,11 +471,41 @@ private boolean endpointsAvailable(ApplicationId id, ZoneId zone, DualLogger log
...
+                var policy = policies.get(new RoutingPolicyId(id, ClusterSpec.Id.from(endpoint.name()), zone));
+                if (policy == null)
+                    throw new IllegalStateException(endpoint + " has no matching policy in " + policies);
+
+                var cNameValue = controller.jobController().cloud().resolveCName(endpointName);
+                if (cNameValue.isEmpty()) {
+                    logger.log(INFO, "CNAME '" + endpointName + "' does not yet point to anything");
+                    return false;
+                }
+                if ( ! cNameValue.get().equals(policy.canonicalName())) {
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: Combine with previous condition? Or is it necessary to treat them differently?

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/TesterCloud.java [hunk 3]
```diff
@@ -29,14 +32,11 @@
...
+    /** Returns the IP address of the given host name, if any. */
+    Optional<String> resolveHostName(HostName hostname);
 
-    /**
-     * Returns whether the given URL is registered in DNS. Always returns true,
-     * as endpoints are not use in this case
-     */
-    default boolean exists(DeploymentId deploymentId) { return true; }
+    /** Returns the host name of the given CNAME, if any. */
+    Optional<HostName> resolveCName(HostName hostName);
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/TesterCloud.java]: Consider renaming to either `resolveCNAME` or `resolveCname` (I think we mostly use the latter: camel-case regardless of actual casing).

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/InternalStepRunner.java]: Could probably collapse this, yes.

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/TesterCloud.java (hunk 3 above)
Comment (jonmv) [reply controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/deployment/TesterCloud.java]: Will do the latter.

**Comment by jonmv** [comment]
PTAL.

**Review by mpolden — APPROVED** [review]
Great!

**Pull request merged**